text/0032-access_command.md

```diff
@@ -0,0 +1,215 @@
+pushed to the registry it will show up in feeds for other users to browse and
+pull.
+```
+
+`$ qri access info me/dataset` for now will only but it will be _very_ helpful for viewing dataset permsissions that control both who can access a dataset and what they can do with it.
+
+### Publish on registry push
+Pushing unencrypted datasets† to the registry _requires_ a dataset be published. To keep the old behaviour of a one-liner publish, pushing to the registry will automatically run publish on the user's behalf. The push command will present this as user feedback:
+
+```
+$ qri push me/some_dataset
+setting dataset access to "published"... done.
+pushing 28 versions of chriswhong/some_dataset...
+X of XX blocks pushed for version 12 August 2019 (Qmss4h552h3j2h33)...
+```
+
+Remotes, on the other hand _can_ accept unpublished datasets. Remote configuration will get a `requirePublish` configuration parameter to get the same registry-like behaviour. Pushing unpublished datasets is aimed at the collaborator use-case, where a datasets are explicitly pushed to trusted colleagues. when a user is pushed an unpublished dataset, it will show up when they run `qri list` locally, but this dataset won't be broadcast back to the network.
+
+† _encrypted datasets don't exist yet._
+
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+### The `Published` field
+For some time we've modelled `published` as a boolean field outside the dataset itself, stored as a logbook operation. We've struggled with this field not capturing the problem of _where_ a dataset's been published. Re-conceiving of `published` as a "permission bit" clarifies the intent of this field. Published now answers "can the existence of this dataset be broadcasted" for the _entire history_ of a dataset. 
+
+### Corner cases
+A few extra bullet points to consider while we're implementing this:
+
+* qri list `--published` should work as it does today, showing only local published datasets
+* We should create a new set of `lib.AccessMethods`, talking to a number of subsystems for access control management.
+* many `publish` tests should move over to the new `push` command
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Access control is really irritating.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+### `listed` and `unlisted`
+Instead of the publish/unpublish terminology, we could use a more direct reference to list visibility. We can't use `list` directly because `qri access list` should be the command for listing all access-controlled datasets (if such a thing turns out to be necessary). An alternative would be to use a different conjucation or tense for list like "listed" & "unlisted". using a past tense is the best I could come up with, and in my opinion `$ qri access listed` isn't very memorable, and feels like your asking about things that have been listed in the past?
```

Comment:
Strong dislike. The term is not as good.

But even if it was the best term, any command or flag that runs that close to another command should always be avoided.

Reply:
Agree here, "listed" is not great, but I think "published" is also ambiguous, especially in the case of working within a network of known peers. How about "visible" vs "hidden", or "public" vs "secret", something along those lines?

Reply:
big fan of "visible" & "hidden"! I'll update the RFC to use those terms instead.

The reason for not "public" and "secret": collides with the statement "all datasets in qri are public".